Sources/App/Features/Sponsors/Models/Middleware/SponsorMiddleware.swift

```diff
@@ -0,0 +1,35 @@
+import Fluent
+
+struct SponsorMiddleware: AsyncModelMiddleware {
```

Comment:
minor: feels like there should be a way to make this solution more generic and not only for sponsors, but we'll tackle that improvement in a future PR.

Reply:
I have ideas for that, but want to discuss as it's more in-depth than sponsors.